CRM/Core/DAO.php

```diff
@@ -2734,6 +2734,15 @@ public static function createSQLFilter($fieldName, $filter, $type = NULL, $alias
             }
             return $escapedCriteria;
 
+          // The "IS" operator is like an alias; it can be rewritten into other criteria.
+          case 'IS':
+          case 'IS NOT':
+            $newFilter = \Civi\Api4\Utils\CoreUtil::rewriteIsCriteria($fieldName, $operator, $criteria);
+            if ($newFilter !== NULL) {
+              return self::createSQLFilter($fieldName, $newFilter, $type, $alias, $returnSanitisedArray);
+            }
+            return NULL;
```

Comment:
Again I'm not sure that adding this to CRM_Core_DAO is a good idea.
It might be better for APIv4 to depart from using the `CRM_Core_DAO::createSQLFilter` function or at least wrap it.
There are several ways in which APIv4 is diverging from the assumptions in this function, and that will be even more so if we implement stuff like string expressions in the WHERE clause, which I do think we should implement soon.